### PR TITLE
[flang][cuda] Avoid generating cuf.data_transfer in OpenACC region

### PR DESCRIPTION
**Pull request by clementval**
`cuf.data_transfer` will be converted to runtime calls to cuda runtime api and these are not supported in device code. assignment in OpenACC region will be handled by the OpenACC code gen so we avoid to generate data transfer on them.

### CODE REVIEW

File: flang/lib/Lower/Bridge.cpp [hunk 1]
```diff
@@ -4380,9 +4380,19 @@ class FirConverter : public Fortran::lower::AbstractConverter {
...
     if (builder.getRegion().getParentOfType<cuf::KernelOp>())
       return true;
+    if (builder.getRegion().getParentOfType<mlir::acc::LoopOp>())
+      return true;
+    if (builder.getRegion().getParentOfType<mlir::acc::KernelsOp>())
+      return true;
+    if (builder.getRegion().getParentOfType<mlir::acc::ParallelOp>())
+      return true;
+    if (builder.getRegion().getParentOfType<mlir::acc::SerialOp>())
+      return true;
```
Comment (jeanPerier) [inline flang/lib/Lower/Bridge.cpp:4395]: Would it make sense to use the ComputeRegionOpInterface to detect those here instead?

Comment (clementval) [reply flang/lib/Lower/Bridge.cpp:4395]: Yeah! I totally forgot that we have this interface now!

**Review by jeanPerier — APPROVED** [review]
LGTM

**Review by razvanlupusoru — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**